### PR TITLE
fix: hydration error after runtime refactor

### PR DESCRIPTION
**Pull request by Nemikolh**
This PR fixes an error at runtime where React complains about an hydration mismatch between what was rendered on the server vs on the client.

This is because during SSR, we do not set the initial commands and thus the `steps` are empty while on the client they are set sooner. I wouldn't have expected that to be the case but the easier here is to just split `runCommands` in two functions:
* `setCommands` which is similar to `setPreviews` and `setTerminalConfiguration`.
* `runCommands` which interact with WebContainer

The error fixed by this PR:

image

### CODE REVIEW

**Review by SamVerschueren — APPROVED** [review]

File: packages/runtime/src/tutorial-runner.ts [hunk 1]
```diff
@@ -71,6 +72,22 @@ export class TutorialRunner {
     private _stepController: StepsController,
   ) {}
 
+  /**
+   *
```
Comment (d3lm) [inline packages/runtime/src/tutorial-runner.ts:76]: Can we add a description here as well?

File: packages/runtime/src/tutorial-runner.ts [hunk 2]
```diff
@@ -71,6 +72,22 @@ export class TutorialRunner {
...
+   * @param commands Commands schema
```
Comment (d3lm) [inline packages/runtime/src/tutorial-runner.ts:77]:
```suggestion
   * @param commands The commands schema.
```

File: packages/runtime/src/tutorial-runner.ts [hunk 3]
```diff
@@ -71,6 +72,22 @@ export class TutorialRunner {
...
+   */
+  setCommands(commands: CommandsSchema) {
+    const newCommands = new Commands(commands);
+    let anyChange = this._changeDetection(commands);
```
Comment (d3lm) [inline packages/runtime/src/tutorial-runner.ts:81]: Newline to separate const from let?

Comment (Nemikolh) [reply packages/runtime/src/tutorial-runner.ts]: Oh actually `let` wasn't needed! I changed it to a `const`.

File: packages/runtime/src/tutorial-runner.ts [hunk 4]
```diff
@@ -71,6 +72,27 @@ export class TutorialRunner {
     private _stepController: StepsController,
   ) {}
 
+  /**
+   * Set the commands to run. This updates the reported `steps` if any have changed.
+   *
+   * This function is safe to call server side.
+   *
+   * To actually run them in webcontainer see `runCommands`.
```
Comment (d3lm) [inline packages/runtime/src/tutorial-runner.ts:80]:
```suggestion
   * To actually run them in WebContainer see `runCommands`.
```

**Review by d3lm — COMMENTED** [review]
Tiny tiny nitpick 🙉 Otherwise LGTM 👍

File: packages/runtime/src/tutorial-runner.ts (hunk 4 above)
Comment (Nemikolh) [reply packages/runtime/src/tutorial-runner.ts:80]: Ah good catch! :pray:

**Review by d3lm — APPROVED** [review]

**Pull request merged**